jmx/src/main/java/org/jboss/as/jmx/model/MBeanInfoFactory.java

```diff
@@ -175,13 +175,9 @@ private OpenMBeanAttributeInfo getAttribute(String name) {
         ModelNode attribute = providedDescription.require(ATTRIBUTES).require(name);
         AttributeAccess access = resourceRegistration.getAttributeAccess(PathAddress.EMPTY_ADDRESS, name);
         if (access == null) {
-            // Check for a bogus attribute in the description that's really a child
-            Set<String> childTypes = resourceRegistration.getChildNames(PathAddress.EMPTY_ADDRESS);
-            if (childTypes.contains(name)) {
```

Comment:
@kabir Yes, please have a look at line 185 https://github.com/wildfly/wildfly-core/blob/d37c460ab49861a82efb5a622a2b0c2d9403cf03/jmx/src/main/java/org/jboss/as/jmx/model/MBeanInfoFactory.java#L185 
It is not possible to create the instance with access being null.

Reply:
Ah I see, I think this check was to cover some corner case, which very possibly isn't relevant any longer. But yes I see now if the access is null, and we don't return due to my original check that it would blow up further down.